File: frontend/rea-investment-fe/src/api/accessible-entities.ts
import type { AxiosInstance } from 'axios';

export interface AccessibleCompany {
  id: number;
  name: string;
}

export interface AccessibleProject {
  id: number;
  name: string;
  company_id: number;
  company_name: string;
}

export interface AccessibleEntitiesResponse {
  companies: AccessibleCompany[];
  projects: AccessibleProject[];
}

export const buildAccessibleEntitiesApi = (httpClient: AxiosInstance) => {
  const getAccessibleEntities = async (): Promise<AccessibleEntitiesResponse> => {
    const response = await httpClient.get<AccessibleEntitiesResponse>('/api/users/account/me/accessible-entities');
    return response.data;
  };

  return Object.freeze({
    getAccessibleEntities
  });
};

File: backend/ilios-server/app/schema/accessible_entities.py
"""API Schema for accessible entities endpoint (context bar)."""

from typing import List

from pydantic import BaseModel, ConfigDict, Field


class AccessibleCompanySchema(BaseModel):
    """Minimal company info for context bar picker."""
    
    model_config = ConfigDict(from_attributes=True)
    
    id: int = Field(examples=[1])
    name: str = Field(examples=["Green Lantern"])


class AccessibleProjectSchema(BaseModel):
    """Minimal project (site) info for context bar picker."""
    
    model_config = ConfigDict(from_attributes=True)
    
    id: int = Field(examples=[1])
    name: str = Field(examples=["Apollo Solar Farm"])
    company_id: int = Field(examples=[1])
    company_name: str = Field(examples=["Green Lantern"])


class AccessibleEntitiesResponse(BaseModel):
    """Response containing all accessible companies and projects for a user."""
    
    companies: List[AccessibleCompanySchema] = Field(default_factory=list)
    projects: List[AccessibleProjectSchema] = Field(default_factory=list)
